gemonlinedb/include/gem/onlinedb/DataSet.h

```diff
@@ -0,0 +1,103 @@
+#ifndef _DataSet_h_
+#define _DataSet_h_
+
+#include <string>
+#include <vector>
+
+#include "gem/onlinedb/ConfigurationTraits.h"
+
+namespace gem {
+    namespace onlinedb {
+
+        // Forward decl.
+        template<class ConfigurationTypeT>
+        class XMLSerializationData;
+
+        /**
+         * @brief Represents a dataset to be stored in the XML file (a
+         *        @c <DATA_SET> tag).
+         */
+        template<class ConfigurationTypeT>
+        class DataSet
+        {
+        public:
+            /**
+             * @brief Convenience typedef.
+             */
+            using ConfigurationType = ConfigurationTypeT;
+
+        private:
+            friend class XMLSerializationData<ConfigurationType>;
+
+            std::string comment, version;
```

Comment:
Style within `cmsgemos` is generally to make `class` variables have a `m_` prefix, to differentiate them from function local variables.
The same is not applied to `struct` `typedefs` (one can easily argue that this is inconsistent, but it is what is currently done)

Reply:
Applied to all private member variables. Public variables and types untouched.

Reply:
> Public variables and types untouched.

Reason for not also applying to public variables?

Reply:
Many things in cmsgemos look like Java to me, so I follow the [Java conventions](https://en.wikipedia.org/wiki/Naming_convention_(programming)#Java). In Java, public member variables and types don't take any prefix. Since private variables are hidden by definition, no convention exists and developers use whatever they like.

Moreover, I could find the following `struct`s in cmsgemos: [[1]](https://github.com/cms-gem-daq-project/cmsgemos/blob/develop/gembase/include/gem/base/GEMApplication.h#L183) [[2]](https://github.com/cms-gem-daq-project/cmsgemos/blob/develop/gembase/include/gem/base/GEMMonitor.h#L182), and none of them uses leading `m_`s. The one exception I found is [`GEMApplication`](https://github.com/cms-gem-daq-project/cmsgemos/blob/develop/gembase/include/gem/base/GEMApplication.h#L211) whose interface doesn't seem to be final anyway.

Note that in this PR, all public member variable belong to structs whose sole functionality is to group sets of related variables together. If at some point I need to add functions, I'll turn them into classes, add encapsulation and `m_` prefixes.

Reply:
OK, I agree that the utilized style is a bit of a mix between various projects I've worked on and various styles I've read through (though when starting this one I tried to apply a consistent set of them...)

> Moreover, I could find the following `struct`s in `cmsgemos`: [1] [2], and none of them uses leading `m_`s.

As mentioned in the initial comment, `structs` are explicitly treated differently, and their member elements are not prefixed with `m_` (or `p_` for a member pointer as you saw in `GEMApplication`)

> Note that in this PR, all public member variable belong to `struct`s

OK, then as per the aforementioned style, everything should be consistent, and I won't belabour the point further